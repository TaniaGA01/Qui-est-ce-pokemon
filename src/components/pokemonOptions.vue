<template>
  <div class="options-container">
    <ul>
      <li 
      v-for="pokemon in pokemons" 
      :key="pokemon.id"
      @click="$emit('selectPokemon', pokemon.id)"><!-- dos argumentos : 1. Nombre del evento (argumento obligatorio) 2. Data (opcional). El valor 1 se envía al padre, el valor dos se envía como argumento a la funcion donde el evento 1 funciona-->
      {{ pokemon.name }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props:{
    pokemons:{ // viene de pokemonPage.vue
      type: Array,
      required: true
    }
  }
}
</script>

<style scoped>
ul {
  list-style-type: none;
  display: contents;
}
li {
  font-size: 1rem;
  background-color: #c75164;
  padding: 5px 10px;
  border-radius: 100px;
  font-weight: bolder;
  color: rgb(255, 255, 255);
  cursor: pointer;
  width: 250px;
  text-transform: uppercase;
  display: grid;
  align-items: center;
  -webkit-box-shadow: 0px 0px 6px 0px rgb(0, 0, 0, 0.3); 
  box-shadow: 0px 0px 6px 0px rgb(0, 0, 0, 0.3);
  transition: 0.2s;
}
li:hover {
  background-color: #962437;
  transform: scale(1.05);
  -webkit-box-shadow: 0px 0px 20px 0px rgb(0, 0, 0, 0.3); 
  box-shadow: 0px 0px 20px 0px rgb(0, 0, 0, 0.3);
}
.options-container {
  margin-top: 25px;
  display: flex;
  justify-content: center;
}
@media (max-width: 991px) {
    ul {
      display: grid;
    }
    li {
      margin-top: 2rem;
      width: auto;
      display: initial;
    }
    .options-container{
        display: initial;
    }
 }
</style>
