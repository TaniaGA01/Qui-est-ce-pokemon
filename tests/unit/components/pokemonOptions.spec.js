import { shallowMount } from "@vue/test-utils"
import pokemonOptions from "@/components/pokemonOptions"
import { pokemons } from '../mocks/pokemons.mock'

describe('pokemonOptions Component', () => {

    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(pokemonOptions, {
            props: {
                pokemons
            }
        })
        
        console.log(pokemons)
    })
    
    test('Match avec snapshot', () => {

        console.log( wrapper.html() )

        expect( wrapper.html() ).toMatchSnapshot()

    })

    test('Afficher les 4 li', () => {
        
        const liTags = wrapper.findAll('li')

        expect(liTags.length).toBe(4)
        
        expect(liTags[0].text() ).toBe('bulbasaur')
        expect(liTags[1].text() ).toBe('ivysaur')
        expect(liTags[2].text() ).toBe('venusaur')
        expect(liTags[3].text() ).toBe('charmander')

    })

    test('Emitir "selection" avec les paramÃ¨tres en cliquant ', () => {
        
        const [li1, li2, li3, li4] = wrapper.findAll('li')

        li1.trigger('click'); // un click manuel --> estimule pour l'expect
        li2.trigger('click'); // un click manuel --> estimule pour l'expect
        li3.trigger('click'); // un click manuel --> estimule pour l'expect
        li4.trigger('click'); // un click manuel --> estimule pour l'expect

        console.log(wrapper.emitted('selectPokemon')) // [[1]] corresponde au pokemon.id

        expect( wrapper.emitted('selectPokemon').length).toBe(4) // emitir 1 seule fois
        expect( wrapper.emitted('selectPokemon')[0]).toStrictEqual([1]) // [1] pokemon.id
        expect( wrapper.emitted('selectPokemon')[1]).toStrictEqual([2])
        expect( wrapper.emitted('selectPokemon')[2]).toStrictEqual([3])
        expect( wrapper.emitted('selectPokemon')[3]).toStrictEqual([4])

    })

})