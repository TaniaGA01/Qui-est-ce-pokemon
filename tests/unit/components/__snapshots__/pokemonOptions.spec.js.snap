// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemonOptions Component Match avec snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>
      <!-- dos argumentos : 1. Nombre del evento (argumento obligatorio) 2. Data (opcional). El valor 1 se envía al padre, el valor dos se envía como argumento a la funcion donde el evento 1 funciona-->bulbasaur
    </li>
    <li>
      <!-- dos argumentos : 1. Nombre del evento (argumento obligatorio) 2. Data (opcional). El valor 1 se envía al padre, el valor dos se envía como argumento a la funcion donde el evento 1 funciona-->ivysaur
    </li>
    <li>
      <!-- dos argumentos : 1. Nombre del evento (argumento obligatorio) 2. Data (opcional). El valor 1 se envía al padre, el valor dos se envía como argumento a la funcion donde el evento 1 funciona-->venusaur
    </li>
    <li>
      <!-- dos argumentos : 1. Nombre del evento (argumento obligatorio) 2. Data (opcional). El valor 1 se envía al padre, el valor dos se envía como argumento a la funcion donde el evento 1 funciona-->charmander
    </li>
  </ul>
</div>
`;
